packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -175,51 +175,57 @@ export class SOQLEditorInstance {
       }
       case MessageType.UI_TELEMETRY: {
         const { unsupported, errors } = e.payload as TelemetryModelJson;
-        if (errors) {
+        const hasErrors = Array.isArray(errors) ? errors.length : errors;
+        const hasUnsupported = Array.isArray(unsupported)
+          ? unsupported.length
+          : unsupported;
+        if (hasErrors) {
           trackError('syntax_error', JSON.stringify(e.payload)).catch(
             console.error
           );
         }
-        if (unsupported) {
+        if (hasUnsupported) {
           // no need to duplicate.  unsupported and errors often both present
-          if (!errors) {
+          if (!hasErrors) {
             trackError('syntax_unsupported', JSON.stringify(e.payload)).catch(
               console.error
             );
           }
-          channelService.appendLine(`This syntax is not yet supported.`);
+          channelService.appendLine(
+            `This syntax is not yet supported in SOQL Builder. Use a text editor.`
```

Comment:
Emily - updated error message

Reply:
This syntax is not yet supported in SOQL Builder. Instead, use a text editor.